base.sh

```diff
@@ -261,18 +246,30 @@ restart () {
 	   crcount=0
     fi
 
-    ## cr_restart
-    ## wait until restart is really done
-    ## nice trick from KennethHoste
+    # resume from checkpoint
+    # wait until resume is actually complete via lockfile
     rm -f "$chkrestat"
     rm -f "$chklock"
     lockfile "$chklock"
-    cr_restart $cr_restart_restore --run-on-success="echo OK > \"$chkrestat\" && rm -f \"$chklock\"" --run-on-failure="echo FAILURE > \"$chkrestat\" && rm -f \"$chklock\"" -f "$chkfile" >& "$chkreout" &
+    # using --new-coordinator doesn't seem to work, so start DMTCP coordinator ourselves as daemon and use that
+    $DMTCP_COORDINATOR --daemon --coord-logfile "$chkdir/coord.log.$$" --coord-port 0 --port-file "$chkdir/$PORTFILE" --ckptdir $chkdir --exit-on-last --interval 0
+    coord_port=$(cat "$chkdir/$PORTFILE")
+    myecho "DMTCP coordinator port: $coord_port"
+    $DMTCP_RESTART --coord-port $coord_port `find $chkdir -name '*.dmtcp'` &
```

Comment:
find might retrun more then one file? add explict `|head -1` or `|tail -1`

Reply:
We actually *need* all `*.dmtcp` files to be listed here...

When checkpointing, DMTCP creates a `*.dmtcp` file for each process, and it recycles `*.dmtcp` files on the next checkpoint.